### DISCUSSION TITLE
[Core Protocol] How are the base fee and the blob base fee trustlessly relayed from Ethereum to OP Mainnet?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [ethereum-optimism/developers#417] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I was reading the docs on [Transaction Fees](https://docs.optimism.io/stack/transactions/fees), and I noticed these statements:

> The L1 Data Fee is most heavily influenced by the Ethereum base fee that is continuously and trustlessly relayed from Ethereum to OP Mainnet.
> 
> ...
> 
> The current Ethereum base fee and/or blob base fee (trustlessly relayed from Ethereum).

Little explanation is provided on how the fees are trustlessly relayed from Ethereum. How does that happen?

**Comment by hash1go** [comment, marked as answer]
The information regarding gas prices is obtained from the predeploy address [GasPriceOracle](https://specs.optimism.io/protocol/predeploys.html#gaspriceoracle). This GasPriceOracle in turn retrieves its information from another predeploy address, [L1Block](https://specs.optimism.io/protocol/predeploys.html#l1block). The L1Block contract is updated through system transactions that are initiated by a system account.

The data from L1 is fetched using an RPC URL to connect to an Ethereum node. Therefore, as I understand it, the system trusts the Ethereum node it references, making it not strictly trustless.
